Foody/Onboarding/Login/LoginViewController.swift

```diff
@@ -6,29 +6,112 @@
+        welcomeLbl.text = "Welcome"
+        loginLbl.text =  "Login your account"
+        loginBttn.layer.cornerRadius = 15
+        registerBttn.layer.cornerRadius = 15
     }
+   
+   
+    @IBAction func rememberMeSwitchChck(_ sender: UISwitch) {
+        let defaults: UserDefaults? = UserDefaults.standard
+        if sender.isOn {
+            defaults?.set(true, forKey: "ISRemember")
+            print("switch is on")
+            }
+            else {
+                defaults?.set(false, forKey: "ISRemember")
+                print("switch is off")
+            }
+        }
+    
+    @IBAction func loginBttnClicked(_ sender: Any) {
+        guard let email = emailTextFld.text else { return }
+        guard let password = passwordTextFld.text else { return }
+        if email != "" && password != "" {
+                Auth.auth().signIn(withEmail: email, password: password) { (result, error) in
+                       if error != nil{
+//                           self.errorMessage(titleInput: "Error!", messageInput: error?.localizedDescription ?? "Try again.")
+                           if !email.isValidEmail(email: email){
+                               self.errorMessage(titleInput: "Error!", messageInput: "Please control the email")
+                           }else if !password.isValidPassword(password: password){
+                               self.errorMessage(titleInput: "Error!", messageInput: "Please control the password")
+                           }
+                       }else{
+                           self.errorMessage(titleInput: "WELCOME!", messageInput: error?.localizedDescription ?? "Success Login.")
+                           UserDefaults.standard.set(true, forKey: "REMEMBER_USER")
+                           print("success login")
+                       }
+                   }
+               }
+        else{
+                   errorMessage(titleInput: "Error!", messageInput: "Enter the username and password")
+               }
+           }
     
+    func errorMessage(titleInput:String, messageInput:String){
+           let alert = UIAlertController(title: titleInput, message: messageInput, preferredStyle: UIAlertController.Style.alert)
+           let okButton = UIAlertAction(title: "OK", style: UIAlertAction.Style.default, handler: nil)
+           alert.addAction(okButton)
+           self.present(alert, animated: true, completion: nil)
+       }
+         
     @IBAction func clickedRegister(_ sender: Any) {
         clickedRegisterCompletion?()
     }
     
-    /*
-    // MARK: - Navigation
+}
+extension String {
 
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
-    }
-    */
+    func isValidEmail(email: String) -> Bool {
```

Comment:
Please make function private.  Search what is different between private and public.